stripe_test.go

```diff
@@ -118,6 +119,71 @@ func TestDo_Retry(t *testing.T) {
 	assert.Equal(t, 2, requestNum)
 }
 
+// Tests client retries when HTTP returns timeout.
+//
+// To run only this test:
+//
+//     go test . -run TestDo_RetryOnTimeout -test.v
+//
+func TestDo_RetryOnTimeout(t *testing.T) {
+	type testServerResponse struct {
+		Message string `json:"message"`
+	}
+
+	timeout := time.Second
+	requestNum := 0
+
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		err := r.ParseForm()
+		assert.NoError(t, err)
+
+		// The body should always be the same with every retry. We've
+		// previously had regressions in this behavior as we switched to HTTP/2
+		// and `Request` became non-reusable, so we want to check it with every
+		// request.
+		assert.Equal(t, "bar", r.Form.Get("foo"))
```

Comment:
Let's strip out this `assert` and just rely on the one in `TestDo_Retry` above (it and the comment above it were copied from there).